libchromaprint.vapi

```diff
@@ -0,0 +1,368 @@
+/*
+ * The MIT License
+ *
+ * Copyright (C) 2017 - Mario Daniel Ruiz Saavedra
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+/**
+ * Chromaprint is a library for generating audio fingerprints, mainly to be used with the [[https://acoustid.org|AcoustID]] service.
+ *
+ * It needs raw audio stream (16-bit signed int) on input. The audio can have any sampling rate and any number of channels. Typically,
+ * you would use some native library for decoding compressed audio files and feed the result into Chromaprint.
+ *
+ * Audio fingerprints returned from the library can be represented either as
+ * base64-encoded strings or 32-bit integer arrays. The base64-encoded strings
+ * are usually what's used externally when you need to send the fingerprint
+ * to a service. You can't directly compare the fingerprints in such form.
+ * The 32-bit integer arrays are also called "raw fingerprints" and they
+ * represent the internal structure of the fingerprints. If you want to
+ * compare two fingerprints yourself, you probably want them in this form.
+ *
+ * ==Generating fingerprints==
+ *
+ * Here is a simple example code that generates a fingerprint from audio samples in memory:
+ *
+ * {{{
+ *
+ * const int sample_rate = 44100;
+ * const int num_channels = 2;
+ *
+ * var ctx = new ChromaContext();
+ *
+ * ctx.start(sample_rate, num_channels);
+ * while (has_more_data) {
+ *     ctx.feed(data, size);
+ * }
+ * ctx.finish();
+ *
+ * string fp =  ctx.get_fingerprint();
+ * print("%s\n", fp);
+ *
+ * }}}
+ *
+ * Note that there is no error handling in the code above. Almost any of the called functions can fail.
+ * You should check the return values in an actual code.
+ */
+[CCode (cheader_filename = "chromaprint.h")]
```

Comment:
You could avoid most `cname` declaration here by setting `lower_case_cprefix` and `cprefix`.

Reply:
Yeah, but in such a small vapi, I can just search for the full name of the original function. If it's required, however, I can change it.

Reply:
I was merely suggesting it in case you didn't know.